modules/dnn/src/layers/convolution_layer.cpp

```diff
@@ -648,6 +656,53 @@ class ConvolutionLayerImpl : public BaseConvolutionLayerImpl
         }
     };
 
+#ifdef HAVE_OPENCL
+    bool forward_ocl(std::vector<Mat*> &inputs, std::vector<Mat> &outputs, std::vector<Mat> &internals)
+    {
+        int group = inputs[0]->size[1] / blobs[0].size[1];
+
+        if (convolutionOp.empty())
+        {
+            OCL4DNNConvConfig config;
+            config.in_shape = {inputs[0]->size[0], inputs[0]->size[1], inputs[0]->size[2], inputs[0]->size[3]};
+            config.out_shape = {outputs[0].size[0], outputs[0].size[1], outputs[0].size[2], outputs[0].size[3]};
+            config.kernel = {kernel.height, kernel.width};
+            config.pad = {pad.height, pad.width};
+            config.stride = {stride.height, stride.width};
+            config.dilation = {dilation.height, dilation.width};
+            config.group = group;
+            config.bias_term = (hasBias()) ? true : false;
+            config.weights_backward = false;
+            config.bias_backward = false;
+
+            convolutionOp = Ptr<OCL4DNNConvSpatial<float> >(new OCL4DNNConvSpatial<float>(config));
+        }
+
+        UMat weights, biases;
+        weights = blobs[0].getUMat(ACCESS_READ);
+        if (hasBias()) biases = blobs[1].getUMat(ACCESS_READ);
```

Comment:
Probably constant values can be pre-allocated and filled into `cv::UMat` during "preparation" stage.